src/pal/src/misc/time.cpp

```diff
@@ -270,6 +275,29 @@ GetTickCount64()
         }
         retval = (ts.tv_sec * tccSecondsToMillieSeconds)+(ts.tv_nsec / tccMillieSecondsToNanoSeconds);
     }
+#elif HAVE_MACH_CLOCK_MONOTONIC
+    {
+        kern_return_t machRet;
+        clock_serv_t clock;
+        mach_timespec_t mts;
+        host_t host = mach_host_self();
+
+        if((machRet = host_get_clock_service(host, SYSTEM_CLOCK, &clock)) != KERN_SUCCESS)
```

Comment:
Thanks for the feedback @jkotas! I updated the benchmark and got the results which are close to the numbers you predicted.

With my changes (no caching) - Average time 2.8 microseconds.
With my changes (with caching) - Average time 0.82 microseconds.
Without my changes (using gettimeofday) - Average time 0.05 microseconds.

So with caching enabled it is around it is around 16.4x slower.

Here is the updated microbench mark

```c#
        const long tccSecondsToMicroSeconds = 1000000;

        [Fact]
        public static void TickCountPerf()
        {            
            int iterations = 10000000;            
            long startTime = GetTimeofDay();

            for (int i = 0; i < iterations; i++)
            {               
                int tick = Environment.TickCount;
                
            }

            long totalTime = GetTimeofDay() - startTime;

            float average = (float)totalTime / iterations;
            Console.WriteLine("totalTime -- {0} usecs", totalTime);
            Console.WriteLine("Average -- {0} usecs", average);
        }

        public static long GetTimeofDay()
        {
            timeval tv = new timeval();           
            int result = gettimeofday(ref tv, IntPtr.Zero);            
            //Console.WriteLine("return value - " + (tv.tv_sec * tccSecondsToMicroSeconds) + tv.tv_usec);
            return (tv.tv_sec * tccSecondsToMicroSeconds) + tv.tv_usec;
        }

        [DllImport("libc")]
        //private static extern int gettimeofday(ref timeval tv, ref timezone tz);
        private static extern int gettimeofday(ref timeval tv, IntPtr tz);
    }

    struct timeval
    {
        public long tv_sec; /* seconds since Jan. 1, 1970 */
        public long tv_usec; /* and microseconds */
    }
```